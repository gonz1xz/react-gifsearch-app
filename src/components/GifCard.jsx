import { useState } from "react"
import PropTypes from 'prop-types'

export const GifCard = ({ url, title }) => {

    const [isHidden, setIsHidden] = useState(false);


    return (
        <div className="card"
            onMouseEnter={() => setIsHidden(true)}
            onMouseLeave={() => setIsHidden(false)}
        >
            <img src={url} alt={title} />
            {isHidden && <p> {title} </p>}
        </div>
    )
}

GifCard.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}


