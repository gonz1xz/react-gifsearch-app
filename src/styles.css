* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(216, 216, 216);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(0, 60, 255);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h1 {
  text-align: center;
  padding: 0 0 10px 0;
}

h2 {
  font-size: 1.2rem;
}

h3 {
  font-size: 1.5rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  position: relative;
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  width: 300px;
  height: 300px;  
  justify-content: center;
  margin: 20px 20px 20px 0;
  overflow: hidden;
}

.card img {
  object-fit: cover;
  width: 100%;
  height: 100%;
}

.card p {
  position: absolute;
  background-color: rgba(0, 0, 0, 0.479);
  color: white;
  width: 100%;
  height: 100%;
  top: 85%;
  left: 0;
  margin: 0 auto;
  text-align: center;
}