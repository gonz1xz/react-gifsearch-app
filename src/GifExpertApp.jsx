import { useState } from 'react';
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

	const [categories, setCategories] = useState([]);

	const onAddCategory = (newCategory) => {

		if (categories.includes(newCategory)) return console.log("Ya existe");
		setCategories([newCategory, ...categories]);
	}


	return (
		<>
			<h1>Search Gifs App</h1>

			<AddCategory
				onNewCategory={onAddCategory}
			/>

			{
				categories.map((category) => (
					<GifGrid key={category} category={category} />
				))
			}

		</>
	)
}