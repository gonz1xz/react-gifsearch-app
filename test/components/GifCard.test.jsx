import { GifCard } from '../../src/components/';
import { render, screen } from '@testing-library/react';

describe('Pruebas en GifCard.jsx', () => {

    const title = 'Saitama';;
    const url = 'https://one-punch.com/saitama.jpg';

    test('Debe coincidir con snapshot', () => {


        const { container } = render(<GifCard title={title} url={url} />);

        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado ', () => {

        render(<GifCard title={title} url={url} />);

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe de mostrar el titulo en el componente', () => {

        // render(<GifCard title={title} url={url} />);

        // expect(screen.getByText(title)).toBeTruthy();

    });




});
