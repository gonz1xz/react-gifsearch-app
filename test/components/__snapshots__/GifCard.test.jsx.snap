// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifCard.jsx Debe coincidir con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
